names/names.py

```diff
@@ -12,17 +12,57 @@
 
 duplicates = []  # Return the list of duplicates in this data structure
 
+class BSTNode:
+    def __init__(self, value):
+        self.value = value
+        self.left = None
+        self.right = None
+        
+    #insert value into tree
+    def insert(self, value):
+        if value < self.value:
+            if self.left is None:
+                self.left = BSTNode(value)
+            else:
+                self.left.insert(value)
+        else:
+            if self.right is None:
+                self.right = BSTNode(value)
+            else:
+                self.right.insert(value)
+                
+    #return True if value present, False if not
+    def contains(self, target):
+        if self.value == target:
+            return True
+        
+        if target < self.value:
+            if not self.left:
+                return False
+            else: 
+                return self.left.contains(target)
+        
+        else:
+            if not self.right:
+                return False
+            else: 
+                return self.right.contains(target)
 # Replace the nested for loops below with your improvements
-for name_1 in names_1:
-    for name_2 in names_2:
-        if name_1 == name_2:
-            duplicates.append(name_1)
+bst = BSTNode(names_1[0])
```

Comment:
Excellent job implementing the BST and using its methods!

Reply:
thank you!